docs/topics/whatsnew1720.md

```diff
@@ -0,0 +1,749 @@
+
+See this [YouTrack issue](https://youtrack.jetbrains.com/issue/KT-53797) for more information about this builder inference update.
+
+## Kotlin/JVM
+
+Kotlin 1.7.20 introduces generic inline classes, adds more bytecode optimizations for delegated properties, and supports
+IR in the kapt stub generating task, making it possible to use all the newest Kotlin features with kapt:
+
+* [Generic inline classes](#generic-inline-classes)
+* [More optimized cases of delegated properties](#more-optimized-cases-of-delegated-properties)
+* [Support for the JVM IR backend in the kapt stub generating task](#support-for-the-jvm-ir-backend-in-the-kapt-stub-generating-task)
+
+### Generic inline classes
+
+> Generic inline classes is an [Experimental](components-stability.md#stability-levels-explained) feature.
+> It may be dropped or changed at any time. Opt-in is required (see details below), and you should use it only for evaluation purposes.
+> We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-52994).
+>
+{type="warning"}
+
+Kotlin 1.7.20 allows the underlying type of JVM inline classes to be a type parameter. The compiler maps it to `Any?` or,
+generally, to the upper bound of the type parameter.
+
+Consider the following example:
+
+```kotlin
+@JvmInline
+value class UserId<T>(val value: T)
+
+fun compute(s: UserId<String>) {} // compiler generates fun compute-<hashcode>(s: Any?)
+```
+
+The function accepts the inline class as a parameter. The parameter is mapped to the upper bound, not the type argument.
+
+To enable this feature, use the `-language-version 1.8` compiler option.
+
+We would appreciate your feedback on this feature in [YouTrack](https://youtrack.jetbrains.com/issue/KT-52994).
+
+### More optimized cases of delegated properties
+
+In Kotlin 1.6.0, we optimized the case of delegating to a property by omitting the `$delegate` field and generating
+immediate access to the referenced property. In 1.7.20, we've implemented this optimization for more cases.
+The `$delegate` field will now be omitted if a delegate is:
+
+* A named object:
+
+  ```kotlin
+  object NamedObject {
+      operator fun getValue(thisRef: Any?, property: KProperty<*>): String = ...
+  }
+  
+  val s: String by NamedObject
+  ```
+  {validate="false"}
+
+* A final `val` property with a backing field and a default getter in the same module:
+
+  ```kotlin
+  val impl: ReadOnlyProperty<Any?, String> = ...
+  
+  class A {
+      val s: String by impl
+  }
+  ```
+  {validate="false"}
+
+* A constant expression, an enum entry, `this`,  or `null`. Here's an example of `this`:
+
+  ```kotlin
+  class A {
+      operator fun getValue(thisRef: Any?, property: KProperty<*>) ...
+   
+      val s by this
+  }
+  ```
+  {validate="false"}
+
+Learn more about [delegated properties](delegated-properties.md).
+
+We would appreciate your feedback on this feature in [YouTrack](https://youtrack.jetbrains.com/issue/KT-53768).
```

Comment:
```suggestion
We would appreciate your feedback on this feature in [YouTrack](https://youtrack.jetbrains.com/issue/KT-23397).
```

Reply:
Seems like I mixed up and left the link to the doc issue instead of the initial issue.